handlers/middlewares.go

```diff
@@ -101,12 +101,13 @@ func (h *Handler) SessionInjectorMiddleware(next func(http.ResponseWriter, *http
 			logrus.Warn("unable to read session from the session persister, starting with a new one")
 		}
 
-		if prefObj == nil {
-			prefObj = &models.Preference{
-				AnonymousUsageStats:  true,
-				AnonymousPerfResults: true,
-			}
-		}
+		// if prefObj == nil {
+		// 	prefObj = &models.Preference{
+		// 		AnonymousUsageStats:  true,
+		// 		AnonymousPerfResults: true,
+		// 	}
+		// }
+		h.checkIfK8SConfigExistsOrElseLoadFromDiskOrK8S(req, user, prefObj, provider)
```

Comment:
Can you please verify it with mesheryctl, Server is already able to detect kubeconfig.

Reply:
@kushthedude when you say `mesheryctl`, do you mean test the update in a containerized deployment? `mesheryctl` doesn't need access to kubeconfig, does it? Meshery server is currently throwing error logs in to the affect of "k8s config is nil".

Reply:
@leecalcote The error only occurs when meshery is launched through mesheryctl, if you try by make run-local the server indeed detects the kubeconfig. This is what is happening on my machine and this is what the issue describes too.

Reply:
@kushthedude makes sense. I experience the same behavior. I think the terms "built and running on local machine" / "running as a local process" and "running in a container" / "running containerized" help clarify the differences in deployment models. It wasn't clear if you were referring to `mesheryctl` as a component in question or as a method of deploying (a deployment model).

Reply:
Ahh, we referred to same thing with different terms 😅

Reply:
😄  I do think that "Does this work when you deploy the change using `mesheryctl`" makes sense, too, because mesheryctl is only capable of targeting Docker as the deployment platform. I'm hopeful that someday soon, this phrasing won't be clear, because hopefully `mesheryctl` will be capable of targeting Kubernetes as the deployment platform.